pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,399 @@
+`
+
+func NewCommandManageNode(f *clientcmd.Factory, commandName, fullName string, out io.Writer) *cobra.Command {
+	cfg := &NodeConfig{}
+
+	cmd := &cobra.Command{
+		Use:   commandName,
+		Short: "Manage node operations: schedulable, evacuate, list-pods",
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+
+			defaultNamespace, err := f.DefaultNamespace()
+			kcmdutil.CheckErr(err)
+
+			_, kc, err := f.Clients()
+			kcmdutil.CheckErr(err)
+
+			mapper, typer := f.Object()
+
+			cfg.DefaultNamespace = defaultNamespace
+			cfg.Kclient = kc
+			cfg.Writer = out
+			cfg.Mapper = mapper
+			cfg.Typer = typer
+			cfg.RESTClientFactory = f.Factory.RESTClient
+			cfg.Printer = f.Printer
+			cfg.Operations = []string{}
+			cfg.NodeNames = []string{}
+
+			if c.Flag("schedulable").Changed {
+				cfg.Operations = append(cfg.Operations, "schedulable")
+			}
+			if c.Flag("evacuate").Changed {
+				cfg.Operations = append(cfg.Operations, "evacuate")
+			}
+			if c.Flag("list-pods").Changed {
+				cfg.Operations = append(cfg.Operations, "list-pods")
+			}
+
+			if len(args) != 0 {
+				cfg.NodeNames = append(cfg.NodeNames, args...)
+			}
+
+			errList := []error{}
+			if errList = cfg.ValidateParams(); len(errList) != 0 {
+				err := fmt.Errorf("%v", errList)
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			err = cfg.RunManageNode()
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", false, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", false, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&cfg.ListPods, "list-pods", false, "List all/selected pods on the node.")
+
+	flags.BoolVar(&cfg.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
+	flags.Var(&cfg.PodSelector, "pod-selector", "Label selector to filter pods on the node. Optional parameters for --evacuate or --list-pods")
+	flags.Var(&cfg.Selector, "selector", "Label selector to filter nodes. Either pass one/more nodes as arguments or use this node selector")
+
+	return cmd
+}
+
+func (cfg *NodeConfig) ValidateParams() []error {
+	errList := []error{}
+	if cfg.Selector.Provided() {
+		if _, err := labels.Parse(cfg.Selector.Value()); err != nil {
+			errList = append(errList, errors.New("--selector=<node_selector> must be a valid label selector"))
+		}
+		if len(cfg.NodeNames) != 0 {
+			errList = append(errList, errors.New("either specify --selector=<node_selector> or nodes but not both"))
+		}
+	} else if len(cfg.NodeNames) == 0 {
+		errList = append(errList, errors.New("either specify --selector=<node_selector> or nodes"))
+	}
+
+	numOps := len(cfg.Operations)
+	if numOps == 0 {
+		errList = append(errList, errors.New("must provide a node operation. Supported operations: --schedulable, --evacuate and --list-pods"))
+	} else if numOps != 1 {
```

Comment:
@deads2k @smarterclayton 
Please look at http://pastebin.test.redhat.com/282982
It has around 11 options and it is hard to figure out which one is actual option to perform operation and which ones are optional params for the selected operation. And also its hard to figure out which optional params are valid to specific operation.
With subcommands, admins may need to read couple of help pages but they can get the command right in the first attempt but with one command, I think it will be difficult.

Let me know if we want to stick with single command for now or to have subcommands